src/vs/workbench/services/workingCopy/common/workingCopyFileService.ts

```diff
@@ -344,19 +377,19 @@ export class WorkingCopyFileService extends Disposable implements IWorkingCopyFi
 		return stats;
 	}
 
-	async delete(resources: URI[], options?: { useTrash?: boolean, recursive?: boolean, undoRedoGroupId?: number, isUndoing?: boolean }, token?: CancellationToken): Promise<void> {
+	async delete(operations: IDeleteOperation[], undoInfo?: IFileOperationUndoRedoInfo, token?: CancellationToken): Promise<void> {
 
 		// validate delete operation before starting
-		for (const resource of resources) {
-			const validateDelete = await this.fileService.canDelete(resource, options);
+		for (const operation of operations) {
+			const validateDelete = await this.fileService.canDelete(operation.resource, { recursive: operation.recursive, useTrash: operation.useTrash });
```

Comment:
Instead of creating a new options bag I could just pass the `operation` object
@bpasero let me know what you prefer.
Samo on line `409`

Reply:
I think it is fine to create a new option bag, the way it is now 👍 . I often prefer to explicitly call out how a method is called to make it clear when reading the code.